script.js

```diff
@@ -1,4 +1,147 @@
+var winCount = 0;
+var lossCount = 0;
+var inputCount = 0;
+var username;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (inputCount != 0 && isInvalidInput(input)) {
+    return "There are only 3 input options, please try again";
+  }
+
+  if (inputCount == 0) {
+    inputCount = inputCount + 1;
+    return saveUser(input);
+  }
+
+  var computer = generateComputerSelection();
+
+  var winStatus = checkWinStatus(input, computer);
+  inputCount = inputCount + 1;
+
+  if (winStatus == "win") {
+    winCount = winCount + 1;
+  }
+  if (winStatus == "loss") {
+    lossCount = lossCount + 1;
+  }
+  return generateOutput(winStatus, input, computer);
+};
+
+// Save user name
+var saveUser = function (input) {
+  username = input;
```

Comment:
maybe you can say hello ${username} to give feedback to the user that the name is saved correctly